spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java

```diff
@@ -151,4 +167,27 @@ TableCatalog tableCatalog() {
       return tableCatalog;
     }
   }
+
+  /**
+   * Closes the procedure's executor service if a new one was created.
```

Comment:
Nit; maybe just but comment can be clearer?  (maybe method link?)

Maybe put a comment on executorService() call and say we need to call this method, as thats the one that would be called first?